operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml

```diff
@@ -0,0 +1,572 @@
+        kind: IngressController
+        metadata:
+          name: eiptestnlb
+          namespace: openshift-ingress-operator
+        spec:
+          endpointPublishingStrategy:
+            loadBalancer:
+              scope: External
+              providerParameters:
+                type: AWS
+                aws:
+                  type: NLB
+                  networkLoadBalancer:
+                    subnets:
+                      ids:
+                      - subnet-0fcf8e0392f0910d5
+                      names:
+                      - subnetA
+                      - subnetB        
+                    eipAllocations:
+                    - eipalloc-1234567890abcdefa
+                    - eipalloc-1234567890abcdefb
+            type: LoadBalancerService
+      expectedError: "IngressController.operator.openshift.io \"eiptestnlb\" is invalid: spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer: Invalid value: \"object\": number of subnets must be equal to number of eipAllocations"
+    - name: Should not allow ingress controller creation if number of subnets names provided is not equal to the number of eipAllocations.
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: IngressController
+        metadata:
+          name: eiptestnlb
+          namespace: openshift-ingress-operator
+        spec:
+          endpointPublishingStrategy:
+            loadBalancer:
+              scope: External
+              providerParameters:
+                type: AWS
+                aws:
+                  type: NLB
+                  networkLoadBalancer:
+                    subnets:                    
+                      names:
+                      - subnetA
+                      - subnetB        
+                    eipAllocations:
+                    - eipalloc-1234567890abcdefa
+            type: LoadBalancerService
+      expectedError: "IngressController.operator.openshift.io \"eiptestnlb\" is invalid: spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer: Invalid value: \"object\": number of subnets must be equal to number of eipAllocations"
+    - name: Should allow to ingress controller creation if number of ids provided is not equal to the number of eipAllocations.
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: IngressController
+        metadata:
+          name: eiptestnlb
+          namespace: openshift-ingress-operator
+        spec:
+          endpointPublishingStrategy:
+            loadBalancer:
+              scope: External
+              providerParameters:
+                type: AWS
+                aws:
+                  type: NLB
+                  networkLoadBalancer:
+                    subnets:
+                      ids:
+                      - subnet-0fcf8e0392f0910d5
+                      names:
+                      - subnetA
+                      - subnetB        
```

Comment:
I think you mistakenly included names here, otherwise this is a duplicate test case.